platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/TestReportGenerator.java

```diff
@@ -16,15 +16,32 @@
 
 package org.gradle.api.internal.tasks.testing;
 
-import org.gradle.internal.operations.BuildOperationExecutor;
-import org.gradle.internal.operations.BuildOperationRunner;
 import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
 
 import java.nio.file.Path;
+import java.util.List;
 
+/**
+ * Shared interface between legacy/JVM and generic test report generators.
+ */
 @NullMarked
 public interface TestReportGenerator {
-    boolean hasResults();
+    /**
+     * A test report generator that does nothing.
+     */
+    TestReportGenerator NO_OP = resultsDirectories -> null;
 
-    void generateReport(BuildOperationRunner operationRunner, BuildOperationExecutor operationExecutor, Path outputDir);
+    /**
+     * Generate a report from the test results in the given results directories, if there are any.
+     *
+     * <p>
+     * Results are placed in the given report directory.
+     * </p>
+     *
+     * @param resultsDirectories the directories containing the test results
+     * @return the path to the main file/directory of the generated report, or {@code null} if no report was generated
```

Comment:
❔ Does this have to allow for not generating a report?  Can we always require at least outputting an empty file to remove the need for nullability?